test/clojure/core_test/special_symbol_qmark.cljc

```diff
@@ -0,0 +1,51 @@
+(ns clojure.core-test.special-symbol-qmark
+  (:require [clojure.test :refer [deftest testing are]]
+            [clojure.core-test.portability #?(:cljs :refer-macros :default :refer) [when-var-exists]]))
+
+(when-var-exists special-symbol?
+  (deftest test-special-symbol?
+
+    (testing "special symbols"
+      (are [arg] (special-symbol? 'arg)
+                 &
+                 .
+                 case*
+                 catch
+                 def
+                 deftype*
+                 do
+                 finally
+                 fn*
+                 if
+                 let*
+                 letfn*
+                 loop*
+                 new
+                 quote
+                 recur
+                 set!
+                 throw
+                 try
+                 var))
+
+    (testing "more special symbols"
+      #?(:cljs    "cljs doesn't have these"
+         :default (are [arg] (special-symbol? 'arg)
```

Comment:
~~Isn't this just doing `(special-symbol? (quote arg))` for all of these?~~

EDIT: I think we can remove this section. Not every dialect will have these specials and we don't know which ones will. But it doesn't give us any increased confidence in the fn working, since the above tests are sufficient.